internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/utils/ConfigParser.java

```diff
@@ -0,0 +1,248 @@
+            properties.load(input);
+        } catch (IOException e) {
+            throw new ConfigParserException("Error occurred in parsing Atuwa URL : " + e);
+        }
+        AtuwaDownloader.setAtuwaBaseURl(properties.getProperty(JSScannerConstants.ATUWA_BASE_URL));
+    }
+
+    /**
+     * Parse each product configuration details.
+     *
+     * @param productName product name.
+     * @return Product
+     * @throws ConfigParserException exception occurred while parsing configuration details.
+     */
+    public static Product parseProductConfiguration(String productName) throws ConfigParserException {
+        String fileName = null;
+        if (productName.contains(JSScannerConstants.IDENTITYSERVER)) {
+            fileName = JSScannerConstants.IS_CONFIG_FILE;
+        } else if (productName.contains(JSScannerConstants.AM)) {
+            fileName = JSScannerConstants.APIM_CONFIG_FILE;
+        } else if (productName.contains(JSScannerConstants.INTEGRATION)) {
+            fileName = JSScannerConstants.EI_CONFIG_FILE;
+        } else if (productName.contains(JSScannerConstants.DAS)) {
+            fileName = JSScannerConstants.DAS_CONFIG_FILE;
+        } else if (productName.contains(JSScannerConstants.STREAMPROCESSOR)) {
+            fileName = JSScannerConstants.SP_CONFIG_FILE;
+        } else if (productName.contains(JSScannerConstants.OB)) {
+            fileName = JSScannerConstants.OB_CONFIG_FILE;
+        } else if (productName.contains(JSScannerConstants.APIM)) {
+            fileName = JSScannerConstants.APIM_CONFIG_FILE;
+        }
+        Properties properties = new Properties();
+        ClassLoader classLoader = ConfigParser.class.getClassLoader();
+        //load a properties file from class path
+        try (InputStream input = classLoader.getResourceAsStream(fileName)) {
+            properties.load(input);
+        } catch (IOException e) {
+            throw new ConfigParserException("Error occurred in parsing product configurations : " + e);
+        }
+
+        Product productDto = new Product();
+        productDto.setProductRepoName(properties.getProperty(JSScannerConstants.GIT_REPO_NAME));
+        productDto.setInputSourceType(properties.getProperty(JSScannerConstants.INPUT_SOURCE_TYPE));
+        List<String> componentList = Arrays.asList(properties.getProperty(JSScannerConstants.COMPONENT_REPO).
+                split(JSScannerConstants.PROPERTIES_FILE_DELIMETER));
+        List<String> versionTagList = Arrays.asList(properties.getProperty(JSScannerConstants.VERSION_TAG_KEY_WORD).
+                split(JSScannerConstants.PROPERTIES_FILE_DELIMETER));
+        HashMap<String, String> repoVersionMapper = new HashMap<>();
+        for (int i = 0; i < componentList.size(); i++) {
+            repoVersionMapper.put(componentList.get(i), versionTagList.get(i));
+        }
+        productDto.setRepoVersionMapper(repoVersionMapper);
```

Comment:
Everything that depends on successful property loading must go inside the try block.